velox/connectors/hive/HiveWriteProtocol.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/WriteProtocol.h"
+
+namespace facebook::velox::connector::hive {
+
+/// Parameters for Hive writers.
+class HiveWriterParameters : public WriterParameters {
+ public:
+  enum class UpdateMode {
+    kNew, // Write files to a new directory.
+    kAppend, // Append files to an existing directory.
+    kOverwrite, // Overwrite an existing directory.
+  };
+
+  HiveWriterParameters(
```

Comment:
Would you document this constructor and explain when the caller should specify writeFileName and writeDirectory? Should there be some checks based on updateMode?

Reply:
It would really be nice to document this constructor. Otherwise, it will be hard for future readers / users of the API to understand how to use it properly.

Reply:
Sorry, missed this comment previously. Moving the documentation from data members to this constructor. Relations between write & target and updateMode might not be straightforward, which is determined by a concrete implementation of WriteProtocol::getWriterParameters(). So might not want to have checks on the updateMode.